### PR TITLE
Yu

### PR DESCRIPTION
**Pull request by northbigpanda**
1. add libevent API in EventLoop
2. add aarch64 libevent cross compile ability in CMakeList.txt in bsp_socket

### CODE REVIEW

File: src/bsp_sockets/CMakeLists.txt [hunk 1]
```diff
@@ -27,9 +28,55 @@ set(SOURCES
...
+
+
+# Manually set LIBEVENT_LIBRARIES and LIBEVENT_LIBRARY_DIRS
+#set(LIBEVENT_LIBRARIES "/home/builder/temp/aarch64/lib/libevent.so")
+#set(LIBEVENT_LIBRARY_DIRS "/home/builder/temp/aarch64/lib")
+#set(LIBEVENT_INCLUDE_DIRS "/home/builder/temp/aarch64/include")
+
+# 检查是否已设置 MY_PKG_CONFIG_PATH 变量
+#if(DEFINED BSP_PKG_CONFIG_PATH)
+#    # 设置环境变量 PKG_CONFIG_PATH
```
Comment (zczjx) [inline src/bsp_sockets/CMakeLists.txt:42]: 路径配的部分不要屏蔽，因为你现在依赖的别人设置的环境，并行编译的顺序是不确定的

File: src/bsp_sockets/IEventLoop.hpp [hunk 2]
```diff
@@ -23,9 +25,13 @@ using ioCallback = std::function<void(std::shared_ptr<IEventLoop> loop, int fd,
 //让当前loop在一次poll循环后执行指定任务
 using pendingFunc = std::function<void(std::shared_ptr<IEventLoop>, std::any)>;
 
+//using libCallback = std::function<void(evutil_socket_t fd, short what, void *arg)>;
+
 struct IOEvent//注册的IO事件
 {
     int mask{0x00};             //EPOLLIN EPOLLOUT
+    struct event *event_for_read{nullptr};  //libevent
+    struct event *event_for_write{nullptr}; //libevent
```
Comment (zczjx) [inline src/bsp_sockets/IEventLoop.hpp:34]: libevent 的在工厂函数删掉

**Review by zczjx — COMMENTED** [review]
昨天有些还没改好

File: src/bsp_sockets/impl/TcpClient.cpp [hunk 3]
```diff
@@ -38,12 +38,15 @@ static void reConnectCallback(std::shared_ptr<IEventLoop> loop, std::any usr_dat
 
 static void connectEventCallback(std::shared_ptr<IEventLoop> loop, int fd, std::any args)
 {
+    //std::cout<<"41行 in  TcpCLient.cpp" << std::endl;
```
[the remark this answers is not in the document]
Comment (northbigpanda) [reply src/bsp_sockets/impl/TcpClient.cpp]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: src/bsp_sockets/IEventLoop.hpp [hunk 4]
```diff
@@ -39,6 +45,9 @@ struct EventLoopParams
 
     struct pollfd m_fds[1024];
     int m_nfds{0};
+
+    struct event_base *m_event_base{nullptr};
```
[the remark this answers is not in the document]
Comment (northbigpanda) [reply src/bsp_sockets/IEventLoop.hpp]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: src/bsp_sockets/IEventLoop.hpp (hunk 2 above)
Comment (northbigpanda) [reply src/bsp_sockets/IEventLoop.hpp]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: src/bsp_sockets/IEventLoop.hpp [hunk 5]
```diff
@@ -11,6 +11,8 @@
 #include <iostream>
 #include <memory>
 #include <poll.h>
+#include <event2/event.h>
+#include <event2/event_struct.h>
```
[the remark this answers is not in the document]
Comment (northbigpanda) [reply src/bsp_sockets/IEventLoop.hpp]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: src/bsp_sockets/CMakeLists.txt [hunk 6]
```diff
@@ -27,9 +28,55 @@ set(SOURCES
...
+#link_directories(${LIBEVENT_LIBRARY_DIRS})
+#link_directories(/home/builder/temp/aarch64/lib)
+
+# Add library directories for libevent
+#target_link_directories(${PROJECT_NAME} PUBLIC ${LIBEVENT_LIBRARY_DIRS})
+
+# Link the libevent library
+#target_link_libraries(${PROJECT_NAME} PUBLIC libevent.so)
+#target_link_libraries(${PROJECT_NAME} PUBLIC ${LIBEVENT_LIBRARIES})
+
```
[the remark this answers is not in the document]
Comment (northbigpanda) [reply src/bsp_sockets/CMakeLists.txt]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: src/bsp_sockets/CMakeLists.txt (hunk 1 above)
Comment (northbigpanda) [reply src/bsp_sockets/CMakeLists.txt]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

File: Dockerfile [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM zczjx/bsp-perf-build-env-x86-base:latest
```
[the remark this answers is not in the document]
Comment (northbigpanda) [reply Dockerfile]: Done, can be found in 92fe5dd1cd6246a18d90c8b60f10bbd9b495f766

**Pull request merged**